testsuite/billing.py

```diff
@@ -0,0 +1,53 @@
+"""APIs for Stripe and Braintree"""
+import braintree
+import stripe
+
+
+# pylint: disable=too-few-public-methods
+class Stripe:
+    """API for Stripe"""
+
+    def __init__(self, api_key):
+        stripe.api_key = api_key
+
+    @staticmethod
+    def read_payment(customer_id):
+        """Read Stripe payment"""
+        return list(filter(lambda x: x['customer'] == customer_id, stripe.PaymentIntent.list()))[0]
+
+    @staticmethod
+    def read_customer_by_account(account):
+        """Read Stripe customer"""
+        return list(filter(lambda x: str(account.entity_id) in x['metadata'].get("3scale_account_reference", []),
+                           stripe.Customer.list()))[0]
+
+    def invoice_assert(self, account, invoice):
+        """Compare 3scale and Stripe invoices"""
+        customer = self.read_customer_by_account(account)
+        stripe_invoice = self.read_payment(customer['id'])
+        invoice_id = stripe_invoice['metadata']['order_id']
+        currency = stripe_invoice['currency']
+        cost = stripe_invoice['amount'] / 100
```

Comment:
This is that kind of code that deserves some comments. These lines are not intuitive